packages/itwin/grouping-mapping-widget/src/widget/components/Mappings.tsx

```diff
@@ -0,0 +1,115 @@
+import type { CreateTypeFromInterface } from "../utils";
+import type { mappingUIDefaultDisplayStrings } from "./MappingsUI";
+import { MappingsUI } from "./MappingsUI";
+
+export type IMappingTyped = CreateTypeFromInterface<Mapping>;
+
+export interface MappingsProps {
+  onClickAddMapping?: () => void;
+  onClickMappingTitle?: (mapping: Mapping) => void;
+  onClickMappingModify?: (mapping: Mapping) => void;
+  displayStrings?: Partial<typeof mappingUIDefaultDisplayStrings>;
+}
+
+const fetchMappings = async (
+  setMappings: React.Dispatch<React.SetStateAction<Mapping[]>>,
+  iModelId: string,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
+  getAccessToken: GetAccessTokenFn,
+  mappingsClient: IMappingsClient
+) => {
+  try {
+    setIsLoading(true);
+    const accessToken = await getAccessToken();
+    const mappings = await mappingsClient.getMappings(accessToken, iModelId);
+    setMappings(mappings.sort((a, b) => a.mappingName.localeCompare(b.mappingName)));
+  } catch (error: any) {
+    handleError(error.status);
+  } finally {
+    setIsLoading(false);
+  }
+};
+
+export interface MappingsOperationsProps extends GroupingMappingApiConfig {
+  mappingClient: IMappingsClient;
+}
+
+export const useMappingsOperations = ({ iModelId, getAccessToken, mappingClient }: MappingsOperationsProps) => {
+  const [showDeleteModal, setShowDeleteModal] = useState<Mapping | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [mappings, setMappings] = useState<Mapping[]>([]);
+  const [isTogglingExtraction, setIsTogglingExtraction] = useState<boolean>(false);
+
+  useEffect(() => {
+    void fetchMappings(setMappings, iModelId, setIsLoading, getAccessToken, mappingClient);
+  }, [getAccessToken, mappingClient, iModelId]);
+
+  const refresh = useCallback(async () => {
+    setMappings([]);
+    await fetchMappings(setMappings, iModelId, setIsLoading, getAccessToken, mappingClient);
+  }, [getAccessToken, mappingClient, iModelId]);
+
+  const toggleExtraction = useCallback(async (mapping: Mapping) => {
+    try {
+      setIsTogglingExtraction(true);
+      const newState = !mapping.extractionEnabled;
+      const accessToken = await getAccessToken();
+      await mappingClient.updateMapping(accessToken, iModelId, mapping.id, {
+        extractionEnabled: newState,
+      });
+    } catch (error: any) {
+      handleError(error.status);
+    } finally {
+      setIsTogglingExtraction(false);
+    }
+  }, [getAccessToken, iModelId, mappingClient]);
+
+  const onDelete = async (mapping: Mapping) => {
+    const accessToken = await getAccessToken();
+    await mappingClient.deleteMapping(accessToken, iModelId, mapping.id);
+    await refresh();
+  };
+
+  return { mappings, isLoading, refresh, toggleExtraction, onDelete, setShowDeleteModal, showDeleteModal, isTogglingExtraction };
+};
+
+export const Mappings = (props: MappingsProps) => {
```

Comment:
"Mappings" and "MappingsUI" do not feel descriptive enough. Certainly "UI" is too general. MappingsComponent / MappingsWrapper might be a good starting point.

Reply:
I tried not renaming the wrapper component to make it less of a breaking change and adding Component at the end of the name feels a little redundant. Instead of `MappingsUI`, what do you thinking about `MappingsView`? The name emphasizes that it's a view, i.e.. it's only concerned with displaying the information.

Reply:
That works for me, we started trending to naming things "views" as opposed to panels, etc.

Reply:
Renamed.